de.fu_berlin.inf.dpp.whiteboard/frontend/app.js

```diff
@@ -23,8 +24,11 @@ class Whiteboard {
 
     ReactDOM.render(
       <Menu toolManager={this.toolManager}
-      canvasResizer={this.canvasResizer}/> ,
-      document.getElementById('menuContainer'));
+        canvasResizer={this.canvasResizer}
+        ref={menu => this.menu = menu} />,
+      document.getElementById('menuContainer'), () => {
+        this.shortcutManager = new ShortcutManager(this.menu);
```

Comment:
Why is it necessary to set the ShortcutManager after the render method is executed?

Reply:
It is certainly not necessary, but it is recommended.
There are some cases where react would create and render the component asynchronously and would return a promise (the JavaScript equivalent of `Future` objects), we want to be sure that the shortcut manager receives the actual menu component. 

[For more info](https://reactjs.org/docs/react-dom.html#render)